### DISCUSSION TITLE
Activate venv installed by pdm automatically

### DISCUSSION THREAD

**Discussion by tvdboom** [pdm-project/pdm#2731] [Q&A] [answered]
Hi,

I am using pdm in a GitHub actions job, where I start installing some dependencies.

![image](https://github.com/pdm-project/pdm/assets/32366550/3105c775-e070-4e17-b087-aab8766aa83b)

But how do I activate the newly installed venv in `.venv`? I get from [the documentation](https://pdm-project.org/latest/usage/venv/#the-location-of-virtualenvs) that the name of the new env contains some hash. How can I get the name programmatically to run `pdm use` for example?

The problem I encounter is that my next command is running the lint command using pre-commit, but I get the following error, which I believe happens because I am not within the venv that was just installed (with `pre-commit`).

![image](https://github.com/pdm-project/pdm/assets/32366550/36d6ba7e-187a-49bd-b1bc-7fd58a299079)

Thanks

**Comment by huxuan** [comment, marked as answer]
You can try `pdm run pre-commit` OR use the `--global` option when running `pdm install`

**Reply by tvdboom** [reply]
that worked.thanks!
